libraries/tlv-account-resolution/src/state.rs

```diff
@@ -263,6 +349,23 @@ impl ExtraAccountMetas {
     }
 }
 
+/// Trait for getting the pubkey of an account regardless of whether it's an
+/// `AccountMeta` or `AccountInfo`
+pub trait ToPubkey {
+    /// Returns the pubkey of the account
+    fn to_pubkey(&self) -> &Pubkey;
+}
+impl ToPubkey for AccountMeta {
+    fn to_pubkey(&self) -> &Pubkey {
+        &self.pubkey
+    }
+}
+impl ToPubkey for AccountInfo<'_> {
+    fn to_pubkey(&self) -> &Pubkey {
+        self.key
+    }
+}
```

Comment:
That's funny, I had `Addressable` but it sounded so strange to me 😂. I'm liking `AccountKeyable` better.

I also wanted to see if you had ideas on a better way to do this, rather than bring in a new trait

Reply:
I thought about it for a bit and couldn't come up with anything great... the only other possibility I had was to pass in `&[&Pubkey]` to the function, but I don't think that's better